tests/resources/helper_functions.py

```diff
@@ -353,3 +354,8 @@ def get_provider_fees() -> Dict[str, Any]:
     }
 
     return provider_fee
+
+
+def approx_from_wei(amount_a, amount_b) -> float:
+    """Helper function to compare amounts in wei with pytest approx function wirth a relative tolerance of 1e-6."""
+    return int(amount_a) / to_wei(1) == approx(int(amount_b) / to_wei(1))
```

Comment:
I think if we convert from `float` to `Decimal` immediately after generating a random value, it is possible to eliminate the need for approximations and this method could be removed.

```python
# `floats` are stored in a fixed number of bits, and therefore are imprecise. 
# The `Decimal` type can be used to reveal how a float literal is approximated in memory:
>>> Decimal(0.000000001)
Decimal('1.0000000000000000622815914577798564188970686927859787829220294952392578125E-9')
# Using `Decimal` we can quantize it to the correct precision
# This should eliminate the need to approximate.
>>> Decimal(0.000000001).quantize(Decimal(10) ** -18)
Decimal('1.000000000E-9')
```